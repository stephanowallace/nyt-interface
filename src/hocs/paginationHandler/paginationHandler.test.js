import React from 'react';
import { shallow } from 'enzyme';
import PaginationHandler from '.';

const baseComponent = () => <div>test</div>;

describe('PaginationHandler', () => {
  const content = [1, 2, 3, 4, 5, 6, 7];

  it('render default state', () => {
    const PaginationHandlerComponent = PaginationHandler(baseComponent);
    const component = shallow(<PaginationHandlerComponent />);
    expect(component).toMatchSnapshot();
  });

  it('setContent', () => {
    const PaginationHandlerComponent = PaginationHandler(baseComponent);
    const component = shallow(<PaginationHandlerComponent />);
    component.instance().setContent(content);
    expect(component.instance().content).toEqual(content);
    expect(component.props().pagination.data.pageNumber).toBe(1);
    expect(component.props().pagination.data.numberOfPages).toBe(1);
  });

  it('setPageNumber', () => {
    const PaginationHandlerComponent = PaginationHandler(baseComponent);
    const component = shallow(<PaginationHandlerComponent />);
    component.props().pagination.setPageNumber(2);
    expect(component.props().pagination.data.pageNumber).toBe(2);
  });

  it('setPageSize', () => {
    const PaginationHandlerComponent = PaginationHandler(baseComponent);
    const component = shallow(<PaginationHandlerComponent />);
    component.props().pagination.setContent(content);
    component.props().pagination.setPageSize(5);
    expect(component.props().pagination.data.pageSize).toBe(5);
    expect(component.props().pagination.data.numberOfPages).toBe(2);
  });
});
