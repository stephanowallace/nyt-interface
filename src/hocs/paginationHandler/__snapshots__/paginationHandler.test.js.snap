// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationHandler render default state 1`] = `
<baseComponent
  pagination={
    Object {
      "data": Object {
        "numberOfPages": 1,
        "pageNumber": 1,
        "pageSize": 10,
        "paginatedContent": Array [
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 0,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 1,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 2,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 3,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 4,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 5,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 6,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 7,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 8,
            "title": "",
            "url": "",
          },
          Object {
            "byLine": "",
            "publishDate": "",
            "short_url": 9,
            "title": "",
            "url": "",
          },
        ],
      },
      "setContent": [Function],
      "setPageNumber": [Function],
      "setPageSize": [Function],
    }
  }
/>
`;
